contrib/go/src/python/pants/contrib/go/tasks/go_thrift_gen.py

```diff
@@ -83,9 +87,28 @@ def synthetic_target_type(self, target):
   def is_gentarget(self, target):
     return isinstance(target, GoThriftLibrary)
 
+  @memoized_method
+  def _validate_supports_more_than_one_source(self):
+    # Support for doing the right thing with multiple files landed in
+    # https://issues.apache.org/jira/browse/THRIFT-3776; first available in 0.10.0
+    if self.get_options().multiple_files_per_target_override:
+      return
+    try:
+      actual_revision = Revision.semver(self._thrift_binary.version)
+    except Exception as e:
+      self.context.log.warn('Assuming that unparseable thrift version `{}` '
+                            'supports more than one source.'.format(self._thrift_binary.version))
```

Comment:
wonder if it might be better to blow up in this case than to make this assumption?